crates/extensions/tedge_health_ext/src/actor.rs

```diff
@@ -0,0 +1,119 @@
+use async_trait::async_trait;
+use tedge_actors::Actor;
+use tedge_actors::CombinedReceiver;
+use tedge_actors::DynSender;
+use tedge_actors::MessageBox;
+use tedge_actors::ReceiveMessages;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::WrappedInput;
+use tedge_api::health::health_status_up_message;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::TopicFilter;
+
+type HealthInputMessage = MqttMessage;
+type HealthOutputMessage = MqttMessage;
+
+pub struct TedgeHealthMonitorActor {
+    health_check_topics: TopicFilter,
+    mqtt_publisher: DynSender<MqttMessage>,
+}
+
+impl TedgeHealthMonitorActor {
+    pub fn new(mqtt_publisher: DynSender<MqttMessage>) -> Self {
+        let health_check_topics = vec!["tedge/health-check", "tedge/health-check/+"]
+            .try_into()
+            .unwrap();
+        Self {
+            health_check_topics,
+            mqtt_publisher,
+        }
+    }
+
+    pub async fn process_mqtt_message(
+        &mut self,
+        message: MqttMessage,
+    ) -> Result<(), anyhow::Error> {
+        if self.health_check_topics.accept(&message) {
+            //Process the mqtt message and send the reply to the health check request message
+            self.mqtt_publisher
+                .send(health_status_up_message("c8y-device-management"))
+                .await?;
+        }
+        Ok(())
+    }
+}
+
+// FIXME: Consider to use a SimpleMessageBox<LogInput,MqttMessage>
+pub struct HealthManagerMessageBox {
```

Comment:
```suggestion
pub struct HealthMonitorMessageBox {
```
Just for consistency